plugins/metrics/parser/CMakeLists.txt

```diff
@@ -8,13 +8,10 @@ include_directories(
 add_library(metricsparser SHARED
   src/metricsparser.cpp)
 
-find_boost_libraries(
-  filesystem
-  log)
 target_link_libraries(metricsparser
   metricsmodel
   util
-  ${Boost_LINK_LIBRARIES})
+  ${Boost_LIBRARIES})
```

Comment:
Are we sure it's okay to link *every* CodeCompass lib that uses *some* boost libraries against **every Boost lib ever** used by *any* CodeCompass target?

Because right now, this will make `libmetricsparser.so` depend on every boost lib we use:
https://github.com/Ericsson/CodeCompass/blob/master/CMakeLists.txt#L13

As opposed to just depending on `libboostlog` and `libboostfilesystem`.

Reply:
I played around with this Boost library issue. I noticed that even in the current master version, too many Boost libraries are imported, because those are dependencies of other Boost libraries. For example if you check what libs are linked agains metrics parser, you can see that it gets "atomic", "thread", "date_time" and many other Boost libs despite its CMakeLists.txt contains only "filesystem" and "log".
So as far as I can see, this new solution is not worse than the previous one.

Reply:
Right, on the new system and new CMake, imported targets do this. Anyways... we can come back to this later when it comes to one plugin depending alone on a "new" boost lib in particular.